state/protocol/badger/mutator.go

```diff
@@ -1047,7 +1048,7 @@ func (m *FollowerState) handleEpochServiceEvents(candidate *flow.Block) (dbUpdat
 				extendingSetup, err := m.epoch.setups.ByID(epochStatus.NextEpoch.SetupID)
 				if err != nil {
 					if errors.Is(err, storage.ErrNotFound) {
-						return nil, fmt.Errorf("unexpected failure to retrieve EpochSetup (id=%x) stored in EpochStatus for block %x: %w",
+						return nil, irrecoverable.NewExceptionf("could not retrieve EpochSetup (id=%x) stored in EpochStatus for block %x: %w",
 							epochStatus.NextEpoch.SetupID, blockID, err)
```

Comment:
same here, I think we can just bubble the error up. The API never documents anything about `storage.ErrNotFound`. To me, this looks like unnecessary logic. 

```suggestion
					return nil, fmt.Errorf("unexpected error retrieving next epoch setup: %w", err)
```